Automattic/jetpack
# Nav Unification: Force classic view on Customizer screens

Fixes https://github.com/Automattic/wp-calypso/issues/55231

#### Changes proposed in this Pull Request:

---

**TL;DR**: This PR forces the classic view (WP Admin) on all the Customizer screens when Nav Unification is enabled.

---

_These changes only affect WP.com sites._

When we removed the global "show wp-admin pages" toggle in https://github.com/Automattic/wp-calypso/pull/54387, we intentionally continued using its value as fallback when a user has still not set a preferred view with the new "Screen Options" tab. See p7DVsv-ch6-p2#comment-36832.

This is problematic on the Customizer and derived screens (such as Widgets or Menus) because we don't render the "Screen Options" tab there due to the different layout, but the global "show wp-admin pages" toggle is still acting on them. That means that users who has active the global toggle, don't have any way now of switching back to the default view.

However, after reviewing all the Customizer screens with @JanaMW27 in both Calypso and WP Admin, we noted that rather than rendering the "Screen Options" tab in these screens, we can actually force the WP Admin views on them, since no functionality is lost (given that all the Calypso views are just iframing the WP Admin views).

#### Jetpack product discussion
N/A.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
- Install Jetpack Beta on an Atomic site and switch to the branch of this PR.
- Go to https://wordpress.com.
- Switch to the Atomic site.
- Make sure that "Appearance > Customizer" opens the WP Admin Customizer.
- Make sure that "Appearance > Widgets" opens the WP Admin Customizer.
- Make sure that "Appearance > Menus" opens the WP Admin Customizer.
- Make sure that "Appearance > Background" opens the WP Admin Customizer.
- Make sure that "Appearance > Additional CSS" opens the WP Admin Customizer.

#### WP.com diffs

In order to unblock the launch of the block editor for widgets on WP.com, the counterpart WP.com changes have been synced manually in different diffs before this PR has been merged:
- D65216-code (deployed in r229917-wpcom).
- D65237-code (deployed in r230015-wpcom).
- D65308-code.
